Updates for build / narrative text

andybrucenet opened the pull request:
Ran all code thru gofmt / golint
Created simple Makefile
Added simple logger for easier debug output
Updated narrative text extraction to improve newline logic. Basically - newlines (paragraphs) are inserted only where there is a blank line in the narrative. This helps to improve the flow of the generated Word document.

Commit:
Prep for PR

Commit:
Ignore temp files / provide simple Makefile

Commit:
Ran golint and gofmt on all code. Reports clean

Commit:
Updates to make golint and gotest work

afeld commented on control/narrative_section.go:
```diff
@@ -58,7 +60,11 @@ func (n narrativeSection) Fill(data opencontrols.Data, control string) (err erro
 	narrative := data.GetNarrative(control, key)
+	if strings.Contains(control, "AU-2") {
+		logger.Debugf("'%s': '%s': %v", control, key, narrative)
+	}
```
I assume this was just for debugging - mind removing this and the similar lines in other files?

afeld commented on fixtures/fixtures.go:
```diff
@@ -41,3 +45,9 @@ func LoadOpenControlFixture() opencontrols.Data {
+var Expect = gomega.Expect
+
+// HaveOccurred - Declarations for Ginkgo DSL
+var HaveOccurred = gomega.HaveOccurred
```
Smart! Minor nitpick: mind moving them to the top, and making them `const`s?

afeld commented on logger/logger.go:
```diff
@@ -0,0 +1,191 @@
+// Fatalf - log and print
+func Fatalf(format string, args ...interface{}) bool {
+	return internalPrint(3, FATAL, format, args...)
+}
```
While useful, this seems like the kind of functionality we can leverage an external package for, right? Just one more thing for us to maintain otherwise.

If we do want to keep it (for now, or longer), please add documentation about how to use it.

afeld commented on mylogger.log:
```diff
@@ -0,0 +1,21 @@
+
+
+fedramp-templater: narrative_section.go:66: DEBUG 'AU-2 (3)': '': No information found for the combination of standard NIST-800-53 and control AU-2 (3)
+
```
Mind removing this file, and adding to the `.gitignore`?

afeld commented on opencontrols/data.go:
```diff
@@ -28,12 +30,44 @@ func LoadFrom(dirPath string) (data Data, errors []error) {
+	}
+	if hasPrevLine {
+		narrativeFinal += "\n"
+	}
```
Mind moving all of this into a `splitLines` function?

afeld reviewed (changes requested):
Thanks so much for contributing! Aside from the comments, please

* [ ] Add tests for the newline logic

Thanks!

Commit:
For PR #59:

1. Remove extraneous debug msgs
2. Move gomega defs to top of fixtures/fixtures.go
3. Remove all the custom logging
4. Add new indirect test for mergeNewLines

andybrucenet commented:
See checkin notes for details - added some more targets to Makefile for easier (local!) testing. Also ran fedramp-templater on my sample application and verified that it continues to work correctly.

Enjoy!

afeld commented:
Looks great! Last thing: mind documenting how to use the (commonly used) Make targets in `CONTRIBUTING.md`?

Commit:
Documented Makefile, includes example for 'debug'

Commit:
Some formatting changes

Commit:
Make 'Makefile' always display as code

andybrucenet commented:
I have been working heavily on compliance-masonry over the past couple days so some text in the Makefile wasn't quite right ;)

CONTRIBUTING.md should look good now.

afeld reviewed (approved):
You rock, thanks!
